# Titanic Survival Prediction
## Chapter 1: Introduction & Business Understanding

*"A data scientist doesn't just analyze data—they ask questions inspired by the world around them."*

---

## Project Motivation

Last night, I watched *Titanic* (1997), one of the most iconic films ever made. Although the movie tells a fictional love story between Jack Dawson and Rose DeWitt Bukater, it is based on the real sinking of the RMS Titanic on April 15, 1912.

While watching the film, several questions came to mind:

- Were women really more likely to survive?
- Did children have a better chance of survival?
- Did being wealthy increase the probability of surviving?
- Were third-class passengers truly trapped below deck?
- Did traveling with family improve or reduce survival chances?

The movie presents these ideas emotionally, but as a data scientist, I wanted to determine whether the historical passenger data supports these narratives.

Instead of relying solely on the movie, this project uses the famous Titanic dataset to explore the characteristics of passengers and identify the factors associated with survival.

This notebook focuses on **Exploratory Data Analysis (EDA)**, where I will investigate the data, identify meaningful patterns, discover relationships between variables, and generate hypotheses that will later be tested using machine learning models.

---

## Business Problem

Imagine being hired by the White Star Line before the Titanic's maiden voyage.

The company wants to understand which passenger characteristics are most strongly associated with survival during a maritime disaster. These insights could be used to improve future safety policies and evacuation procedures.

Our objective is to analyze historical passenger information and ultimately build a machine learning model capable of predicting whether a passenger would survive based on their characteristics.

---

## Problem Statement

Can we predict whether a passenger survived the Titanic disaster using information such as:

- Age
- Gender
- Passenger Class
- Ticket Fare
- Family Members
- Cabin Information
- Embarkation Port

---

## Type of Machine Learning Problem

This is a **Supervised Machine Learning** problem because the historical outcomes are already known.

Specifically, it is a **Binary Classification** problem.

Target Variable:

**Survived**

- 0 = Passenger did not survive
- 1 = Passenger survived

---

## Project Workflow

This project will follow the complete Data Science lifecycle.

1. Business Understanding
2. Exploratory Data Analysis (Current Notebook)
3. Data Cleaning
4. Feature Engineering
5. Data Preprocessing
6. Machine Learning Model Development
7. Model Evaluation
8. Hyperparameter Tuning
9. Feature Importance Analysis
10. Conclusions

---

## Questions I Want the Data to Answer

Watching the movie inspired several hypotheses that I would like to validate using data.

### Hypothesis 1

Women had a higher survival rate than men.

---

### Hypothesis 2

Children were more likely to survive.

---

### Hypothesis 3

Passengers traveling in First Class had a much greater chance of survival.

---

### Hypothesis 4

Passengers who paid higher ticket fares had better survival rates.

---

### Hypothesis 5

Passengers traveling with family had different survival rates than passengers traveling alone.

---

### Hypothesis 6

Cabin location (deck level) may have influenced survival.

---

### Hypothesis 7

Passenger characteristics interact with one another, meaning survival was influenced by multiple factors rather than a single variable.

---

## Success Criteria

By the end of this project I aim to:

✔ Understand the dataset

✔ Discover hidden patterns

✔ Build meaningful features

✔ Train multiple machine learning models

✔ Explain which variables were most important for predicting survival

✔ Compare whether the historical data supports the story portrayed in the movie

---

Let's begin by importing the necessary Python libraries.

In [1]:
import warnings 
warnings.filterwarnings('ignore')

#Data Manupilation
import numpy as np
import pandas as pd

#data visualizaton
import matplotlib.pyplot as plt
import seaborn as sns

#plot settings
plt.style.use('ggplot')
sns.set_style('whitegrid')

#Display settings
pd.set_option('display.max_columns', None)
pd.set_option('display.max_row', 100)

print('Libraries imported successfully!!!')

Libraries imported successfully!!!


Why These Libraries?

This project relies on several widely used Python libraries in data science.

| Library | Purpose |
|---------|---------|
| Pandas | Loading and manipulating structured data |
| NumPy | Numerical computations |
| Matplotlib | Basic visualizations |
| Seaborn | Statistical data visualization |
| Warnings | Suppress unnecessary warning messages |

These libraries form the foundation of most data science and machine learning projects.

In [5]:
#Loading the dataset

df=pd.read_csv('Titanic-Dataset.csv')
print('Dataset imported successfully')



Dataset imported successfully


## About the Dataset

The Titanic dataset contains demographic and travel information for passengers aboard the RMS Titanic.

Each row represents one passenger, while each column describes a specific characteristic such as age, ticket class, gender, or fare.

The dataset also includes the historical outcome of each passenger, indicating whether they survived or perished during the disaster.

This outcome serves as the target variable for our machine learning model.

In [6]:
df.head(5)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


## First Impression

At first glance, the dataset contains a mix of numerical and categorical variables.

Some variables are straightforward, such as:

- Age
- Fare
- Gender

Others require additional processing before they can be used in a machine learning model.

Examples include:

- Name
- Cabin
- Ticket

These columns may contain valuable information that can later be transformed into useful features.

In [7]:
# Displaying Random Samples

df.sample(10, random_state=42)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
709,710,1,3,"Moubarek, Master. Halim Gonios (""William George"")",male,NaN,1,1,2661,15.2458,NaN,C
439,440,0,2,"Kvillner, Mr. Johan Henrik Johannesson",male,31.0,0,0,C.A. 18723,10.5000,NaN,S
840,841,0,3,"Alhomaki, Mr. Ilmari Rudolf",male,20.0,0,0,SOTON/O2 3101287,7.9250,NaN,S
720,721,1,2,"Harper, Miss. Annie Jessie ""Nina""",female,6.0,0,1,248727,33.0000,NaN,S
39,40,1,3,"Nicola-Yarred, Miss. Jamila",female,14.0,1,0,2651,11.2417,NaN,C
290,291,1,1,"Barber, Miss. Ellen ""Nellie""",female,26.0,0,0,19877,78.8500,NaN,S
300,301,1,3,"Kelly, Miss. Anna Katherine ""Annie Kate""",female,NaN,0,0,9234,7.7500,NaN,Q
333,334,0,3,"Vander Planke, Mr. Leo Edmondus",male,16.0,2,0,345764,18.0000,NaN,S
208,209,1,3,"Carr, Miss. Helen ""Ellen""",female,16.0,0,0,367231,7.7500,NaN,Q
136,137,1,1,"Newsom, Miss. Helen Monypeny",female,19.0,0,2,11752,26.2833,D47,S


## Why Look at Random Samples?

Viewing only the first few rows can sometimes be misleading because datasets may be sorted.

Random sampling provides a better understanding of the diversity of passengers in the dataset and helps identify unusual values or inconsistencies early in the analysis.

In [12]:
# Dataset Shape

rows, cols = df.shape

print(f"Number of Rows: {rows}")
print(f"Number of Columns: {cols}")

Number of Rows: 891
Number of Columns: 12


## Dataset Size

The dataset contains **891 passengers** and **12 variables**.

Although this is a relatively small dataset by modern standards, it is ideal for learning the complete machine learning workflow because it includes:

- Missing values
- Categorical variables
- Numerical variables
- Text features
- Feature engineering opportunities
- A binary classification target

Its manageable size also allows us to visualize relationships easily while still presenting many of the challenges encountered in real-world datasets.